storage/s3.go

```diff
@@ -126,13 +126,72 @@ func (s *S3) Get(ctx context.Context, path string) (io.ReadCloser, error) {
 		Bucket: &s.cfg.Bucket,
 		Key:    &path,
 	}
+
 	out, err := s.s3client.GetObject(ctx, input)
 	if err != nil {
 		return nil, errors.Wrapf(err, "fail to get S3 object %v", path)
 	}
 	return out.Body, nil
 }
 
+// GetWithRetries function downloads the object from S3 and writes it to the
+// writer. It returns the number of bytes written to the writer.
+//
+// It implements a retry mechanism to handle if connection is closed before
+// the end of the download.
+func (s *S3) GetWithRetries(ctx context.Context, path string, writer io.Writer) (int64, error) {
```

Comment:
suggestion(non-blocking): I think that in Go, method writing things to a `io.Writer` are named with the word `Write` somewhere
```suggestion
func (s *S3) WriteWithRetries(ctx context.Context, path string, writer io.Writer) (int64, error) {
```

Reply:
Actually it's because I wanted to keep the scope of the lib, we're doing a s3Client.Get(), to get a file from S3, doing s3lcient.Write would be misleading on the API usage, we're looking to download a file from remote